snapcraft/_store.py

```diff
@@ -281,3 +282,83 @@ def download(snap_name, channel, download_path, arch):
         raise RuntimeError(
             'Failed to download {} at {} (mismatched SHA)'.format(
                 snap_name, download_path))
+
+
+def _get_snap_data(name, store):
+    search_results = store.cpi.search_package(name, 'stable', None)
+    if search_results is None:
+        raise RuntimeError(
+            'Snap {name} cannot be found'.format(
+                name=name))
+    details_url = search_results['_links']['self']['href']
+    details = store.cpi.get(details_url).json()
+    if details.get('result') == 'error':
+        raise RuntimeError(
+            'Error obtaining details for snap {}: {}'.format(
+                name, ', '.join(details['errors'])))
+    return details
+
+
+def gated(snap_name):
```

Comment:
I was going with the convention of functions being called the same as the command. If you guys like that better, no problem for me.